src/github/provider.py

```diff
@@ -144,22 +145,8 @@ def fetch_pr(self, pr_number: int) -> PullRequest:
         Returns:
             PullRequest: The `PullRequest` object containing detailed information about the PR.
         """
-        pr = self.api.pulls.get(pr_number)
+        pr = self.api.pulls.get(repo, pr_number)
         return self._to_PullRequest(pr)
-    
-    def list_prs(self, per_page: int, n_pages: int) -> list[PullRequest]:
```

Comment:
This method isn't used anywhere. Also, it's not very useful unless it has `repo` and some filters.

Reply:
it is used in the previous pipeline. 
let's remove it!

Reply:
Where is it still used? I couldn't find it.

Reply:
Oh I mean it is not in used now.
We can remove it!